src/cache_client.rs

```diff
@@ -159,6 +160,52 @@ impl CacheClient {
         request.send(self).await
     }
 
+    /// Flushes the cache with the given name.
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The name of the cache to be flushed of data.
+    ///
+    /// # Examples
+    /// Assumes that a CacheClient named `cache_client` has been created and is available.
+    /// ```
+    /// # fn main() -> anyhow::Result<()> {
+    /// # use momento_test_util::create_doctest_cache_client;
+    /// # tokio_test::block_on(async {
+    /// use momento::requests::cache::flush_cache::FlushCache;
+    /// # let (cache_client, cache_name) = create_doctest_cache_client();
+    ///
+    /// match cache_client.flush_cache(cache_name.to_string()).await {
+    ///     Ok(_) => println!("Flushed cache: {}", cache_name),
+    ///     Err(e) => eprintln!("Error flushing cache: {}", e),
+    /// }
+    /// # Ok(())
+    /// # })
+    /// # }
+    /// ```
+    /// You can also use the [send_request](CacheClient::send_request) method to delete a cache using a [FlushCacheRequest]:
+    /// ```no_run
+    /// # fn main() -> anyhow::Result<()> {
+    /// # use momento_test_util::create_doctest_cache_client;
+    /// # tokio_test::block_on(async {
+    /// use momento::requests::cache::flush_cache::FlushCache;
+    /// use momento::requests::cache::flush_cache::FlushCacheRequest;
+    /// # let (cache_client, cache_name) = create_doctest_cache_client();
+    ///
+    /// let flush_cache_request = FlushCacheRequest::new(cache_name.to_string());
+    ///
+    /// match cache_client.send_request(flush_cache_request).await {
+    ///     Ok(_) => println!("Flushed cache: {}", cache_name),
+    ///     Err(e) => eprintln!("Error flushing cache: {}", e),
```

Comment:
I think it would help to see a match here on the error for `NOT_FOUND` vs everything else